source/client-side-encryption/tests/README.rst

```diff
@@ -0,0 +1,220 @@
+         "documents": [ { "_id": 1 } ]
+     }
+
+  inserts on the default collection, but
+
+  .. code:: json
+
+      {
+          "name": "insertOne",
+          "object": { "database": "db", "collection": "other" }
+          "documents": [ { "_id": 1 }]
+      }
+
+  inserts in the ``db.other`` collection.
+
+- The addition of the ``mongocryptd_expectations``, for APM events coming from mongocryptd.
+
+
+Each YAML file has the following keys:
+
+- ``database_name`` and ``collection_name``: The default database and collection
+  to use in operations. This can be overridden by a per operation ``object``.
+
+- ``encryption_setup``: Setup that should occur for each test.
+
+  - ``collections``: Describes collections that should be created and prepopulated.
+
+    - ``database``: The database for the collection to create.
+
+    - ``collection``: The name of the collection to create.
+
+    - ``data``: A list of documents, or a file path, to insert. If this is an encrypted
+      collection (specified by the jsonSchema) the data should be inserted *without*
+      automatic encryption (i.e. create a client without configuring encryption to
+      insert this data).
+
+    - ``remoteSchema``: A document or file path for the schema that should be configured remotely on the server.
+
+    - ``external``: If true, create this collection on the external key vault.
+
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some or all of the following fields:
+
+  - ``description``: The name of the test.
+
+  - ``skipReason``: Optional, string describing why this test should be
+    skipped.
+
+  - ``clientOptions``: Optional, parameters to pass to MongoClient().
+
+    - ``clientSideEncryptionOptions``
+
+      - ``kmsProvider``: Fixed as "aws-kms".
+
+      - ``kmsProviderOptions``: Optional.
+
+        - ``uri``: Optional. If set to "localhost:12345", use a local mock KMS service. (TODO: mock KMS not available yet.) 
+
+      - ``extraOptions``:
+
+      - ``schemas``: An optional array of schemas for local configuration.
+
+        - ``database``: The database name for the target collection.
+
+        - ``collection``: The name of the target collection.
+
+        - ``schema``: The schema document.
+
+      - ``useRemoteSchemas``: An optional boolean. If true, schemas are retrieved from the MongoDB server using listCollections.
+
+  - ``kmsPreCmd``: Optional, a command to send to the mock KMS server
+    before the test starts.
+
+  - ``kmsPostCmd``: Optional, a command to send to the mock KMS server
+    after the test ends.
+
+  - ``mockupcryptdPostCmd``: Optional, a command to send to the mock
+    mongocryptd after the test ends.
+
+  - ``operations``: Array of documents, each describing an operation to be
+    executed. Each document has the following fields:
+
+    - ``name``: The name of the operation on ``object``.
+
+    - ``object``: If present, indicates the target collection.
```

Comment:
This is a deviation from other spec tests where `object` is a string (e.g. "collection", "session0"). I assume that the FLE tests require the ability to use different collection objects for tests within the same YAML/JSON file.

Can we come up with an alternative syntax that won't conflict with future goals for a unified format? `operation.collectionNamespace` alongside `operation.collectionOptions` might do the trick. Then, `object` can remain consistent with other specs and you can likely omit the field entirely in your YAML/JSON files since it should default to "collection".

Reply:
I think we can get away with just a string value. If we need to do operations on other collections, I could add a "collection0", "collection1". I tentatively changed it.